### PR TITLE
Feature/update autoware lanelet2 ros interface

### PR DESCRIPTION
**Pull request by adev4a**
# PR Details
This PR adds updates the autoware_lanelet2_ros_interface package to be usable with ros2.
An additional change is made to amathutils_lib that allows the ros2 part of the package to be called correctly.

Note-
While the contents of the unit tests were not changed as a whole, one of the unit tests in autoware_lanelet2_ros_interface was modified so it can pass, it appears that the original implementation before this change is failing. 
This change has only been made in the ros2 version.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [ ] I have added any new packages to the sonar-scanner.properties file
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: common/amathutils_lib/package.xml [hunk 1]
```diff
@@ -13,7 +13,6 @@
   <buildtool_depend condition="$ROS_VERSION == 2">ament_cmake_auto</buildtool_depend>
   
   <build_depend>carma_cmake_common</build_depend>
-  <build_depend condition="$ROS_VERSION == 2">ament_auto_cmake</build_depend>
```
Comment (msmcconnell) [inline common/amathutils_lib/package.xml:16]: Why is this no longer needed?

File: common/autoware_lanelet2_ros_interface/CMakeLists.txt [hunk 2]
```diff
@@ -32,98 +30,174 @@ find_path(PUGIXML_INCLUDE_DIRS
...
-  autoware_msgs
-  geometry_msgs
-  visualization_msgs
-  roslint
-)
 
-set(ROSLINT_CPP_OPTS "--filter=-build/c++14")
-roslint_cpp()
+if(${ROS_VERSION} EQUAL 1)
+  message("Building ros1")
```
Comment (msmcconnell) [inline common/autoware_lanelet2_ros_interface/CMakeLists.txt]: If you want to keep this message then it should be set to the debug log level

File: common/autoware_lanelet2_ros_interface/CMakeLists.txt [hunk 3]
```diff
@@ -32,98 +30,174 @@ find_path(PUGIXML_INCLUDE_DIRS
...
+    add_rostest_gtest(query-test test/test_query.test test/src/test_query.cpp)
+    target_link_libraries(query-test ${catkin_LIBRARIES} autoware_lanelet2_ros_interface_lib)
+    add_rostest_gtest(utilities-test test/test_utilities.test test/src/test_utilities.cpp)
+    target_link_libraries(utilities-test ${catkin_LIBRARIES} autoware_lanelet2_ros_interface_lib)
+
+  endif()
+
+else() #ROS2
+
+  message("Building ros2")
```
Comment (msmcconnell) [inline common/autoware_lanelet2_ros_interface/CMakeLists.txt]: ditto

File: common/autoware_lanelet2_ros_interface/CMakeLists.txt [hunk 4]
```diff
@@ -32,98 +30,174 @@ find_path(PUGIXML_INCLUDE_DIRS
...
+  ## Find dependencies using ament auto
+  find_package(ament_cmake_auto REQUIRED)
+  ament_auto_find_build_dependencies()
+
+  find_package(lanelet2_extension REQUIRED)
+
+  # find_package(PkgConfig)
```
Comment (msmcconnell) [inline common/autoware_lanelet2_ros_interface/CMakeLists.txt]: delete?

**Review by msmcconnell — CHANGES_REQUESTED** [review]

File: common/amathutils_lib/package.xml (hunk 1 above)
Comment (adev4a) [reply common/amathutils_lib/package.xml:16]: Looks like its already there on line 13, also there's a typo

File: common/autoware_lanelet2_ros_interface/CMakeLists.txt (hunk 2 above)
Comment (adev4a) [reply common/autoware_lanelet2_ros_interface/CMakeLists.txt]: I can remove these, they dont seem helpful merged in

**Review by msmcconnell — APPROVED** [review]

**Pull request merged**